### ISSUE TITLE
Screen reader not announcing parameter docs/changing overload docs in parameter hints

### ISSUE THREAD

**Issue by webczat** [microsoft/vscode#110431]
From what I know parameter hints/signature help allows me to cycle through parameters of a method and see their documentation, probably including documentation for actual specific method overloads.
However, when using a screenreader and invoking parameter hints, pressing arrows just reads "<type> <parameter name> hint" and nothing else. At least this is what happens when using the c# extension/omnisharp.
I can't quite confirm but briefly looking at code suggests it actually does provide the needed documentation, so it's just not read or not shown by default, and no way to show it via keyboard.
cc: @isidorn

**Comment by isidorn** [comment]
Related issue: https://github.com/microsoft/vscode/issues/90874
We read only the hint of the paramter hints via an aria alert. A real fix would be to pass focus into the paramter hints, however I hit different issues trying to fix that.

So for this issue I suggest that we not only read the hint but also paramter docs. At the end.
@webczat let me know what you think

Just reading " hint" should not happen - unless a method has no paramters maybe?

**Comment by webczat** [comment]
well it would be useful to know if the selected overload has no parameters or has no more or... and what are docs? just docs for the parameter, or actual selected overload documentation? is the latter also displayed?

**Comment by webczat** [comment]
ping, any status update?

**Comment by isidorn** [comment]
No, no update. However I want to look into this, though we are wrapping up this milestone and next milestone I am on vacation. So assigning to August.

**Comment by isidorn** [comment]
I have pushed a change for this, that when parameter hints are announced we also include the docs at the end.
@webczat Try it out with vscode insiders from tomorrow and let me know how it goes, thanks!

**Comment by webczat** [comment]
okay, that works as in, reads the help for the parameter. however was wondering how that looks for a sighted person? for example how a sighted person can see documentation for a different method overload? isn't that feature related to it?

**Comment by isidorn** [comment]
@webczat thanks for trying it out. Based on your comment adding verified label.

Here's how parameter hints look: once you start typing a rectangular hover appears above the cursor, the top half of the rectangle has the parameters, and the bottom half has the documentation.
So I think the experience is similar, since the documentation is at the end.

Let me know what you think. Thanks!

**Comment by webczat** [comment]
yeah the reading order is correct. what I meant was is there documentation for overloads displayed, or only to parameters? I haven't done extensive testing but two overloads of the same method may have different main non parameter docs. So I was wondering if cycling through parameters also displayed documentation for other overloads, not only the documentation for parameter itself.

**Comment by isidorn** [comment]
@webczat all the comments from all the overloads in the parameter hints are appended to the docs at the end. As the user changes overloads the docs are always the same. Maybe that could be improved fyi @jrieken

**Comment by webczat** [comment]
ahh, the only thing I could usually get was docs for first overload or I am doing something wrong.

**Comment by isidorn** [comment]
@webczat good catch. I was not aware that both the `activeParameter` and the `signature` can have `documentation`. Now I am appending both, first the `activeParameter` and then the `documentation` of the signature at the end. So please try it again from tomorrow's vscode insiders and let me know how it goes. Thanks!

**Comment by isidorn** [comment]
I agree. The good thing is that it is at the end so if it gets too chatty should be easy to skip it. Though looking forward to feedback.
